core/lib/snapshots_applier/src/lib.rs

```diff
@@ -373,16 +409,34 @@ impl SnapshotRecoveryStrategy {
 
     async fn create_fresh_recovery_status(
         main_node_client: &dyn SnapshotsApplierMainNodeClient,
+        snapshot_l1_batch: Option<L1BatchNumber>,
     ) -> Result<SnapshotRecoveryStatus, SnapshotsApplierError> {
-        let snapshot_response = main_node_client.fetch_newest_snapshot().await?;
+        let (l1_batch_number, is_latest) = match snapshot_l1_batch {
```

Comment:
`is_latest` is also used for error context, which I consider more important, because the context of the error IMO *is* substantially different depending on whether the user has specified a snapshot (if they did, it's probably user's fault that a snapshot is not found; if they didn't, it's probably a bug on the main node).